bloomberg/TypeScript
# (WIP) Private name class field transformation

## Description of Changes

- Transforms class properties with private names into WeakMap declarations
- Transforms private name accesses into calls to a helper function which accesses the appropriate emitted WeakMap.
- Transforms private name mutations (except unary operators) into calls to a helper function which accesses the appropriate emitted WeakMap.

## Outside the Scope of this PR

- [Transformation of mutating unary operators (`++`, `--`) - both prefix & postfix](https://github.com/joeywatts/TypeScript/pull/1)
- [Transformation of call expressions involving private names](https://github.com/joeywatts/TypeScript/pull/2)
- [Transformation of destructuring assignments involving private names](https://github.com/joeywatts/TypeScript/pull/3)

## Steps to Test

- Build the compiler (`npm run build:compiler`)
- Run tests (`npm test`).
- Check compiler output of any example programs using ES private properties.
